### ISSUE TITLE
File Upload prolem

### ISSUE THREAD

**Issue by thepresident83** [projectsend/projectsend#674]
## Type

Please complete with an X between the brackets to indicate the type of submission.
This is a:

[x] Bug report

[] Feature request

### Description of the problem

I've created several Clients and Uploader System Users.
When I try to upload a newer file, file has been uploaded correctly (I seen 100% uploaded) and later I see an empty screen said "Upload files". I can't set any Title for uploaded file and if I check on Client "Manage File" tab I cannot see file.
Moreover uploaded file hasn't saved on /upload directory.

!

Please be as clear as possible

### List the steps to reproduce the issue

1. Access Project Send as Client or System User with Uploader role
2. Upload a file

### Environment

* ProjectSend version: r1053
* php version: PHP 5.4.16  (I've same issue on PHP 7.1)
* MySQL version: MariadB 5.5.52 (I've tried on Mysql 5.5+ version, same issue)

### Additional information

_[Optional]_ Write any additional information that you believe might be useful to reproduce the issue

On Apache error_log I seen following log:

[Fri Feb 08 15:01:32.879271 2019] [:error] [pid 3716] [client 2.228.125.126:13584] PHP Warning:  touch(): Utime failed: Permission denied in /var/www/html/projectsend/includes/timthumb/timthumb.php on line 197, referer: https://xxxxx/my_files/

### Related PR

_[Optional]_ If you have submitted a PR related to this issue, please paste the link here

___

### IMPORTANT

Before submitting an issue, please [do a search on the issues page](https://github.com/ignacionelson/ProjectSend/issues) to make sure that the problem hasn't been reported already.

**Comment by lmsilva** [comment]
@thepresident83 

The second problem (about the permission denied) can be solved by giving write permissions to the FILE_CACHE_DIRECTORY used in timthumb.php (like 197).

Make sure this directory is owned by www-data:
root@lmsilva-VirtualBox:/var/www/html# ls -ld ./includes/timthumb/cache
drwxr-xr-x 2 www-data www-data 4096 Apr 11  2018 ./includes/timthumb/cache
root@lmsilva-VirtualBox:/var/www/html# 

As for the first problem you reported, I think that is a bug in the current stable release r1053.
I bumped into a very similar problem where the file would fail to upload every few attempts, and would result in the exact same behavior you are describing (with an empty page, just with the title, instead of the usual screen that lets you specify a friendly file name and description).

Maybe try this...
- in upload-process-form.php
-- find this code (around line 225):
if (!in_array($new_filename,$urls_db_files)) {
   $add_arguments['add_to_db'] = true;
}
-- simply add this after that code snippet:
$add_arguments['add_to_db'] = true;

Basically, according to my debugging, some times the new filename does not get found in that $urls_db_files array, so the argument 'add_to_db' never gets set, so the file doesn't get uploaded.

Not exactly sure if this is the problem you are experiencing (given you said no file gets created in /upload/files/), but maybe this will help.

If the upload problem is happening for all uploads, check permissions on /upload/files/, maybe www-data just doesn't have access to write to that folder (just like that other problem with access to the cache folder).

**Comment by ignacionelson** [comment]
There are checks in place now that let you know if PS does not have write permissions. Marking as solved.
